dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.Orchestration;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Semantic Kernel callable function abstract base class.
+/// </summary>
+public abstract class KernelFunction
+{
+    /// <summary>
+    /// Name of the function. The name is used by the function collection and in prompt templates e.g. {{pluginName.functionName}}
+    /// </summary>
+    public string Name { get; protected set; }
```

Comment:
What's the reason for these properties having a `protected set` accessor? It looks like they're initialized by the constructor... do derived implementations also need to set them directly?

Reply:
This is needed by the Planner implementation but I do want to fix this if possible. Task to track this: https://github.com/microsoft/semantic-kernel/issues/3605